File: Mobile/Demol/src/entity/PublicationProfil.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entity ; 


import java.util.Date;

/**
 *
 * @author xagta1
 */

public class PublicationProfil {

   
    private Integer id;
   
    private Date date;

    private String contenu;
   
    private String picpath;
   
    
    private Profile idProfil;

    public PublicationProfil() {
    }

    public PublicationProfil(Integer id) {
        this.id = id;
    }

    public PublicationProfil(Integer id, Date date, String contenu, String picpath) {
        this.id = id;
        this.date = date;
        this.contenu = contenu;
        this.picpath = picpath;
    }

    public PublicationProfil(Date date, String contenu, String picpath, Profile idProfil) {
        this.date = date;
        this.contenu = contenu;
        this.picpath = picpath;
        this.idProfil = idProfil;
    }
    
    

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getContenu() {
        return contenu;
    }

    public void setContenu(String contenu) {
        this.contenu = contenu;
    }

    public String getPicpath() {
        return picpath;
    }

    public void setPicpath(String picpath) {
        this.picpath = picpath;
    }

  

    public Profile getIdProfil() {
        return idProfil;
    }

    public void setIdProfil(Profile idProfil) {
        this.idProfil = idProfil;
    }

    @Override
    public int hashCode() {
        int hash = 0;
        hash += (id != null ? id.hashCode() : 0);
        return hash;
    }

    @Override
    public boolean equals(Object object) {
        // TODO: Warning - this method won't work in the case the id fields are not set
        if (!(object instanceof PublicationProfil)) {
            return false;
        }
        PublicationProfil other = (PublicationProfil) object;
        if ((this.id == null && other.id != null) || (this.id != null && !this.id.equals(other.id))) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "PublicationProfil{" + "id=" + id + ", date=" + date + ", contenu=" + contenu + ", picpath=" + picpath + ", idProfil=" + idProfil + '}';
    }

   
    
}

File: Web/src/Pidev/ProfileBundle/Entity/PublicationProfil.php
<?php

namespace Pidev\UserBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use Vich\UploaderBundle\Mapping\Annotation as Vich;
use Symfony\Component\HttpFoundation\File\File;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * PublicationProfil
 *
 * @ORM\Table(name="publication_profil", indexes={@ORM\Index(name="id_profil", columns={"id_profil"})})
 * @ORM\Entity
 * @Vich\Uploadable
 */
class PublicationProfil
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date", type="date", nullable=false)
     */
    private $date;

    /**
     * @var string
     *
     * @ORM\Column(name="contenu", type="text", length=255, nullable=false)
     */
    private $contenu;


    /**
     * NOTE: This is not a mapped field of entity metadata, just a simple property.
     *
     * @Vich\UploadableField(mapping="Publication", fileNameProperty="picpath")
     *
     * @var File
     */
    private $imageFile;

    /**
     * @var string
     *
     * @ORM\Column(name="picpath", type="text", length=255, nullable=false)
     */
    private $picpath;




    /**
     * @var \Profil
     *
     * @ORM\ManyToOne(targetEntity="Profil")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_profil", referencedColumnName="id")
     * })
     */
    private $idProfil;

    /**
     * @return int
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param int $id
     */
    public function setId($id)
    {
        $this->id = $id;
    }

    /**
     * @return \DateTime
     */
    public function getDate()
    {
        return $this->date;
    }

    /**
     * @param \DateTime $date
     */
    public function setDate($date)
    {
        $this->date = $date;
    }

    /**
     * @return string
     */
    public function getContenu()
    {
        return $this->contenu;
    }

    /**
     * @param string $contenu
     */
    public function setContenu($contenu)
    {
        $this->contenu = $contenu;
    }

    /**
     * @return string
     */
    public function getPicpath()
    {
        return $this->picpath;
    }

    /**
     * @param string $picpath
     */
    public function setPicpath($picpath)
    {
        $this->picpath = $picpath;
    }

    /**
     * @return \Profil
     */
    public function getIdProfil()
    {
        return $this->idProfil;
    }

    /**
     * @param \Profil $idProfil
     */
    public function setIdProfil($idProfil)
    {
        $this->idProfil = $idProfil;
    }



    /**
     * @param File|null $image
     * @return $this
     */
    public function setImageFile(File $image = null)
    {
        $this->imageFile = $image;

        if ($image) {
            // It is required that at least one field changes if you are using doctrine
            // otherwise the event listeners won't be called and the file is lost
            $this->updatedAt = new \DateTimeImmutable();
        }

        return $this;
    }

    /**
     * @return File|null
     */
    public function getImageFile()
    {
        return $this->imageFile;
    }

}

